File: Bioinformatics Stronghold/006_Counting_Point_Mutations.py
#!/usr/bin/env python3

from sys import exit, argv

def main():
    if argv[1]:
        fileName = argv[1]
    else:
        exit("No file provided")

    with open(argv[1], 'r') as F:
        sequences = F.read().upper().split('\n')

    print(hammingDist(sequences[0], sequences[1]))


def hammingDist(seq1, seq2):
    diff = 0
    for nt1, nt2 in zip(seq1, seq2):
        if nt1 != nt2:
            diff += 1
    return(diff)


if __name__ == '__main__':
    main()

File: Bioinformatics Stronghold/006_Counting_Point_Mutations.pl
#!/usr/bin/perl
use warnings;
use strict;

my $fileName = shift or die "No file provided\n";

open(F, $fileName) or die "File not found\n";
my @sequences = <F>;
close(F);
chomp(@sequences);

print(hammingDist($sequences[0], $sequences[1]));

exit;

sub hammingDist {
    my ($seq1, $seq2) = @_;
    my $diff;

    if (length($seq1) != length($seq2)) {
        die "Input sequences have different length\n";
    }

    for (my $i = 0; $i < length($seq1); $i++){
        if ((substr($seq1, $i, 1)) ne (substr($seq2, $i, 1))) {
            $diff++;
        }
    }
    return($diff);
}